delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/query/DefaultViewsQuery.scala

```diff
@@ -0,0 +1,99 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.query
+
+import akka.http.scaladsl.model.Uri
+import ch.epfl.bluebrain.nexus.delta.kernel.database.Transactors
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.client.ElasticSearchClient
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.config.ElasticSearchViewsConfig
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.permissions
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.query.ElasticSearchQueryError.{AuthorizationFailed, ElasticSearchClientError}
+import ch.epfl.bluebrain.nexus.delta.sdk.acls.AclCheck
+import ch.epfl.bluebrain.nexus.delta.sdk.acls.model.AclAddress.{Project => ProjectAcl}
+import ch.epfl.bluebrain.nexus.delta.sdk.identities.model.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.model.BaseUri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.{AggregationResult, SearchResults}
+import ch.epfl.bluebrain.nexus.delta.sdk.views.View.IndexingView
+import ch.epfl.bluebrain.nexus.delta.sourcing.Predicate
+import io.circe.JsonObject
+import monix.bio.{IO, UIO}
+
+/**
+  * Allow to list resources from the default elasticsearch views
+  */
+trait DefaultViewsQuery[Result, Aggregate] {
```

Comment:
I made the search and aggregate result generic because returning json in tests makes it difficult to correctly assert things and because the main purpose of this class is in the end to properly filter the views the user has access to